botorch/acquisition/objective.py

```diff
@@ -472,6 +472,8 @@ def __init__(
         """
         super().__init__(objective=objective)
         self.constraints = constraints
+        if type(eta) != Tensor:
+            eta = [eta for _ in range(len(constraints))]
         self.register_buffer("eta", torch.as_tensor(eta))
```

Comment:
Nit: You can use `torch.full` to construct the tensor in one step.
```suggestion
            eta = torch.full(len(constraints), eta)
        self.register_buffer("eta", eta)
```